### DISCUSSION TITLE
Thank you!

### DISCUSSION THREAD

**Discussion by psychosquirrel85** [rhymeswithmogul/aprs-weather-submit#17] [General]
I created bash script for my weather station. My weather station uploads data to wunderground. I then use my API key from wunderground and jq to download that data and format it to fit the parameters you pass via your program. The cronjob runs every 1 minute.  The end result is the following on aprs.fi

![image](https://github.com/user-attachments/assets/86f09521-c481-4347-b97e-07e3c940413d)

Here's the script:

```shell
#!/bin/bash

#Authentication
#Weather Underground
API_KEY="<Weather Underground API Key>"
PWS_ID="<PWS Station ID from wunderground>"

#APRS-IS
call="<HAM Radio Callsign>"
index="<APRS SSID>"
aprs_is_url="<APRS-IS URL>"
aprs_is_port="10152"
aprs_is_user="<APRS-IS Username>"
aprs_is_pass="<APRS-IS Passcode>"
comment="$call - Home Weather Station"

#Gather Weather Underground JSON data from the personal weather station and build an array
response=$(curl -s "https://api.weather.com/v2/pws/observations/current?stationId=$PWS_ID&format=json&units=e&apiKey=$API_KEY")
wx_values=$(echo "$response" | jq -r '.observations[].lat,.observations[].lon,.observations[].imperial.elev,.observations[].imperial.pressure,.observations[].solarRadiation,.observations[].imperial.precipTotal,.observations[].humidity,.observations[].imperial.temp,.observations[].winddir,.observations[].imperial.windSpeed,.observations[].imperial.windGust')
IFS=$'\n' read -r -d '' -a wx_array <<< "$wx_values"

#Expand the array to individual variables for use with "aprs-weather-submit"
lat=${wx_array[0]}
lon=${wx_array[1]}
alt=${wx_array[2]}
baro=${wx_array[3]}
lum=${wx_array[4]}
rain=${wx_array[5]}
rh=${wx_array[6]}
temp=${wx_array[7]}
wdir=${wx_array[8]}
wind=${wx_array[9]}
gust=${wx_array[10]}

#Upload to APRS-IS
aprs-weather-submit -k $call-$index -n $lat -e $lon -A $alt -b $baro -L $lum -P $rain -h $rh -t $temp -c $wdir -S $wind -g $gust -M \"$comment\" -I $aprs_is_url -o $aprs_is_port -u $aprs_is_user -d $aprs_is_pass
```

**Comment by rhymeswithmogul** [comment]
Thank you for using it! It looks like aprs.fi doesn't recognize the voltage parameter, so that's why you have that junk data before your comment (in green).

**Reply by psychosquirrel85** [reply]
Ohhh So that's what this means? But I'm not passing the -V flag for the voltage report.

**Reply by rhymeswithmogul** [reply]
> Ohhh So that's what this means? But I'm not passing the -V flag for the voltage report.

Sorry!  `-X` is radiation.  Omit that and APRS.fi should behave.  I don't remember if it handles altitude cleanly, so if you still see the `/A=` part afterwards, leave off the altitude.  In writing this app, I've found out that APRS.fi has quite a few small bugs!

**Reply by psychosquirrel85** [reply]
I omitted the -X but I'm still seeing the "junk"

![image](https://github.com/user-attachments/assets/62d8238a-f1c5-429a-aaf5-9ab25443dfc6)

**Reply by rhymeswithmogul** [reply]
That's weird.  Let me see your new command line.  Just the call to `aprs-weather-submit` will suffice.

**Reply by psychosquirrel85** [reply]
aprs-weather-submit --callsign KF5SQL-5 --latitude 45.654 --longitude -68.7 --altitude 388 -b 29.54 -L 28.8 -P 0.02 -h 81 -t 46 -c 180 -S 7 -g 7 -I noam.aprs2.net -o 10152 -u aprsuser -d aprskey

**Reply by rhymeswithmogul** [reply]
Oh, and don't feel bad about omitting `-X`.  When it refers to radiation, it doesn't mean UV radiation.  That'd be luminosity.  `-X` is for radioactive radiation, measured by a Geiger counter.  (I'm not sure how *that* wound up in the APRS specification, but it did.)

**Reply by rhymeswithmogul** [reply]
That is weird.  Let me play with this on my end, and I'll let you know if you discovered a bug.

**Reply by psychosquirrel85** [reply]
Thank you for checking. I thought it was working fine lol
